cachi2/core/package_managers/yarn/resolver.py

```diff
@@ -0,0 +1,116 @@
+"""
+Resolve the dependency list for a yarn project.
+
+It also performs the necessary validations to avoid allowing an invalid
+project to keep being processed.
+"""
+import json
+import logging
+from typing import NamedTuple
+
+from cachi2.core.models.sbom import Component
+from cachi2.core.package_managers.yarn.locators import Locator, parse_locator
+from cachi2.core.package_managers.yarn.utils import run_yarn_cmd
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
+
+
+class Cache(NamedTuple):
+    """Cache information for a package."""
+
+    checksum: str
+    path: str
+
+
+class Package(NamedTuple):
+    """A package listed by the yarn info command.
+
+    See the output for 'yarn info -AR --json --cache'.
+
+    Note that the attributes present in the json vary a little depending on the
+    type of dependency.
+    """
+
+    cache: Cache
+    locator: Locator
+    version: str
+
+    @classmethod
+    def from_info_string(cls, info: str) -> "Package":
+        """Create a Package from the output of yarn info."""
+        data = json.loads(info)
+        cache = data["children"]["Cache"]
+
+        return Package(
+            cache=Cache(checksum=cache["Checksum"], path=cache["Path"]),
+            locator=parse_locator(data["value"]),
+            version=data["children"]["Version"],
+        )
```

Comment:
TODO: is children.Version, children.Cache.Checksum, children.Cache.Path always present and non-null?